### PR TITLE
fix: build

### PR DESCRIPTION
**Pull request by rodrigopavezi**
## Problem
It was no building on deployment environment
```
2.891   ▲ Next.js 14.2.19
2.893   - Environments: .env
2.894 
2.895    Linting and checking validity of types ...
28.31    Creating an optimized production build ...
92.24 Segmentation fault (core dumped)
```

## Changes
Add nixpacks configuration for fixing the building issues

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration file to streamline the setup, installation, build, and execution of Node.js applications with Python and C/C++ tools.
	- Added structured phases for setup, installation, building, and starting the application.
	- Defined environment variables to optimize application performance and behavior.

These changes enhance the deployment process and improve application management for developers.

### CODE REVIEW

File: nixpacks.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  'npm install --legacy-peer-deps --no-optional'
+]
+
+[phases.build]
+cmds = [
+  'npm run build'
+]
+
+[start]
+cmd = 'npm start'
```
[the remark this answers is not in the document]
Comment (rodrigopavezi) [reply nixpacks.toml:19]: I will not change that as the build is finally working

File: nixpacks.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+nixPkgs = ['nodejs_18', 'python3', 'gcc', 'gnumake']
+aptPkgs = ['build-essential', 'python3']
+
+[phases.install]
+cmds = [
+  'npm install -g npm@latest',
+  'npm cache clean --force',
+  'rm -rf node_modules package-lock.json .next',
+  'npm install --legacy-peer-deps --no-optional'
+]
```
[the remark this answers is not in the document]
Comment (rodrigopavezi) [reply nixpacks.toml:11]: I will not change that as the build is finally working

[the remark this answers is not in the document]
Comment (MantisClone) [reply nixpacks.toml:11]: @rodrigopavezi can you add a "FIXME" comment to the code pointing out that `--legacy-peer-deps` should be removed if possible (and any other "FIXMEs" you think would be prudent).

**Review by MantisClone — APPROVED** [review]

**Comment by rodrigopavezi** [comment]
Github Action is not running. I will open another PR

**Pull request closed without merging**